packages/design-system/src/components/MonthPicker/MonthPicker.tsx

```diff
@@ -0,0 +1,217 @@
+   */
+  locale?: string;
+  /**
+   * Additional classes to be added to the root element.
+   */
+  className?: string;
+  /**
+   * Applies the "inverse" UI theme
+   */
+  inversed?: boolean;
+  /**
+   * Variation string to be applied to buttons. See [Button component]({{root}}/components/button/#components.button.react)
+   */
+  buttonVariation?: ButtonVariation;
+  /**
+   * Label for the field
+   */
+  label: React.ReactNode;
+  /**
+   * Enable the error state by providing an error message.
+   */
+  errorMessage?: React.ReactNode;
+  /**
+   * Additional classes to be added to the error message
+   */
+  errorMessageClassName?: string;
+  /**
+   * Location of the error message relative to the field input
+   */
+  errorPlacement?: MonthPickerErrorPlacement;
+  /**
+   * Additional hint text to display
+   */
+  hint?: React.ReactNode;
+  /**
+   * Text showing the requirement ("Required", "Optional", etc.). See [Required and Optional Fields]({{root}}/guidelines/forms/#required-and-optional-fields).
+   */
+  requirementLabel?: React.ReactNode;
+  /**
+   * Array of month numbers, where `1` is January, and any month included
+   * is disabled for selection.
+   */
+  disabledMonths?: number[];
+  /**
+   * Array of month numbers, where `1` is January, and any month included
+   * is selected. This will render a read-only field. If the field should
+   * be mutable, use `defaultSelectedMonths`.
+   */
+  selectedMonths?: number[];
+  /**
+   * Array of month numbers, where `1` is January, and any month included
+   * is selected by default. Sets the initial checked state for the 12 month
+   * checkboxes. Use this for an uncontrolled component; otherwise, use the
+   * `selectedMonths` property.
+   */
+  defaultSelectedMonths?: number[];
+  /**
+   * A callback function that's invoked when a month's checked state is changed.
+   * Note: This callback is not called when a month is selected or deselected
+   * via the "Select all" or "Clear all" buttons – use the `onSelectAll` and
+   * `onClearAll` event handlers for those instances.
+   */
+  onChange?: (event: React.ChangeEvent<HTMLInputElement>) => any;
+  onSelectAll?: () => any;
+  onClearAll?: () => any;
+  /**
+   * The text for the "Select all" button for internationalization
+   */
+  selectAllText?: string;
+  /**
+   * The text for the "Clear all" button for internationalization
+   */
+  clearAllText?: string;
+}
+
+export const MonthPicker = (props: MonthPickerProps) => {
+  const months = getMonthNames(props.locale);
+  const monthsLong = getMonthNames(props.locale, false);
+  const isControlled = props.selectedMonths !== undefined;
+  const [selectedMonthsState, setSelectedMonthsState] = useState(props.defaultSelectedMonths ?? []);
+  const selectedMonths = isControlled ? props.selectedMonths : selectedMonthsState;
+  const disabledMonths = props.disabledMonths ?? [];
+
+  function handleChange(event: ChangeEvent<HTMLInputElement>) {
```

Comment:
For consistency, consider changing this to arrow function notation

Reply:
I honestly like the `function` notation better in TypeScript, but if that's how we want to do it, I'm open

Reply:
I think there can be different room for different notations as long as we're consistent about what we use when, especially since they technically behave differently in JavaScript. To me, there's no functional need to make these arrow functions since there is no need for a `this` object to be referenced in any of them. Functions defined like this also get hoisted, so order doesn't matter. I'd listen to an argument, though, about everything being closed to mutation by using `const` everywhere inside these functions...because technically one can re-defined these functions as I've written them.

Reply:
+1 for hoisting powers.